server/src/main/java/org/candlepin/policy/js/entitlement/EntitlementRules.java

```diff
@@ -215,9 +230,26 @@ else if (log.isDebugEnabled()) {
         return filteredPools;
     }
 
-    private Consumer getHost(Consumer consumer) {
-        Consumer host = consumer.hasFact("virt.uuid") ? consumerCurator.getHost(
-            consumer.getFact("virt.uuid"), consumer.getOwner()) : null;
+    /**
+     * Similar to consumerCurator's getHost but here we are ensuring that the owners we search are actually
+     * sharing with this consumer.
+     *
+     * @param consumer
+     * @param pools
+     * @return
+     */
+    private Consumer getHost(Consumer consumer, List<Pool> pools) {
+        if (!consumer.hasFact("virt.uuid")) {
+            return null;
+        }
+        List<Owner> potentialOwners = new ArrayList<Owner>(Arrays.asList(consumer.getOwner()));
```

Comment:
Ah.  Can't really make it a set because the owners get passed in to ConsumerCurator's `getHost` method as varargs.  I did that to maintain backwards compatibility with existing calls and to allow for a `getHost(consumer)` call that automatically searches across all orgs.